blocks/layout-grid/src/grid/edit.js

```diff
@@ -545,80 +515,10 @@ function MaybeDisabledEdit( props ) {
 }
 
 export default compose( [
-	withDispatch( ( dispatch, ownProps, registry ) => ( {
-		/**
-		 * Update all child Column blocks with a new vertical alignment setting
-		 * based on whatever alignment is passed in. This allows change to parent
-		 * to overide anything set on a individual column basis.
-		 *
-		 * @param {string} verticalAlignment the vertical alignment setting
-		 */
-		updateAlignment( verticalAlignment ) {
-			const { clientId, setAttributes } = ownProps;
-			const { updateBlockAttributes } = dispatch( 'core/block-editor' );
-			const { getBlockOrder } = registry.select( 'core/block-editor' );
-
-			// Update own alignment.
-			setAttributes( { verticalAlignment } );
-
-			// Update all child Column Blocks to match
-			const innerBlockClientIds = getBlockOrder( clientId );
-			innerBlockClientIds.forEach( ( innerBlockClientId ) => {
-				updateBlockAttributes( innerBlockClientId, {
-					verticalAlignment,
-				} );
-			} );
-		},
-		updateColumns( previous, columns, columnValues ) {
-			const { clientId } = ownProps;
-			const { replaceBlock } = dispatch( 'core/block-editor' );
-			const { getBlocks } = registry.select( 'core/block-editor' );
-			const innerBlocks = getColumnBlocks(
-				getBlocks( clientId ),
-				previous,
-				columns
-			);
-
-			// Replace the whole block with a new one so that our changes to both the attributes and innerBlocks are atomic
-			// This ensures that the undo history has a single entry, preventing traversing to a 'half way' point where innerBlocks are changed
-			// but the column attributes arent
-			const blockCopy = createBlock(
-				ownProps.name,
-				{
-					...ownProps.attributes,
-					...columnValues,
-					className: removeGridClasses(
-						ownProps.attributes.className
-					),
-				},
-				innerBlocks
-			);
-
-			replaceBlock( clientId, blockCopy );
-		},
-		setPreviewDeviceType( type ) {
-			const { __experimentalSetPreviewDeviceType } = dispatch(
-				'core/edit-post'
-			);
-
-			__experimentalSetPreviewDeviceType( type );
-		},
-	} ) ),
-	withSelect( ( select, { clientId } ) => {
-		const { getBlockOrder, getBlockCount, getBlocksByClientId } = select(
-			'core/block-editor'
-		);
-		const { __experimentalGetPreviewDeviceType = null } = select(
-			'core/edit-post'
-		);
-
-		return {
-			columns: getBlockCount( clientId ),
-			columnAttributes: getBlockOrder( clientId ).map(
-				( innerBlockClientId ) =>
-					getBlocksByClientId( innerBlockClientId )[ 0 ].attributes
-			),
-			previewDeviceType: __experimentalGetPreviewDeviceType(),
-		};
-	} ),
+	withUpdateAlignment(),
+	withUpdateColumns(),
+	withSetPreviewDeviceType(),
+	withColumns(),
+	withColumnAttributes(),
+	withPreviewDeviceType(),
```

Comment:
Is this change to reuse the code or is it because of something else?

Reply:
Yes the change is to reuse the code. So that there is more shared code.